_posts/2020-12-01-infrastructure-migration-finished.md

```diff
@@ -0,0 +1,318 @@
+- Bug-tracker
+- Mail
+- Maillists
+- IRC bots
+- Developers-space
+- Content Delivery Network (just called "mirrors" back in the days)
+- NoAI tournament servers
+- Centralized accounts
+- Subversion
+
+And I am sure that is not a complete list.
+We did weird stuff, not all ideas survived, but boy, we were busy.
+By 2007, most things were either invented or there was no interest in developing new services.
+
+## 11 years later
+
+In 2018, nothing much had changed on the infrastructure side.
+There were some minor additions, some tuning, but nothing really.
+BaNaNaS was still running the same software as it was in 2007.
+The centralized accounts still had "coming soon" on the profile page.
+
+### Is this a bad thing, you ask?
+Well, yes and no.
+From a stability point of view, this was perfect.
+Our services could handle the weirdest things .. for example:
+- We once got [posted on slashdot](https://games.slashdot.org/story/10/04/06/0519206/OpenTTD-100-Released), which is called the slashdot-effect.
+- Once, a [popular YouTube channel](https://www.youtube.com/playlist?list=PLz1xmRxpXfWvEefxXiUo0C8stV6TAsKss) played OpenTTD, doubling our player-base overnight.
+- And many more (smaller) events.
+
+What this means is that many people come to your website, download the game.
+This sounds cool, but it also increases the load on the services.
+Often what you see in these cases, that things crack, and break.
+But .. not on openttd.org.
+The services survived such insane increase in traffic.
+To put it in numbers: we went from between 5 to 15 requests per second, to between 60 and 200 requests per seconds, depending on the event.
+This is an insane amount of additional traffic to handle.
+But, openttd.org could handle it.
+Except for the time a user found a bug in our code that crashed the software .. during such event.
+That was fun :P
+
+### So why the no?
+Well, to give an example: our main website was running Django 1.2, heavily modified.
+This was released in 2007.
+13 years ago.
+There are known bugs for Django 1.2.
+We patched those we knew about, but .. this became increasingly more difficult.
+This means the chances of someone finding a security-issue in any of the services increased more and more.
+This is an accident waiting to happen.
+Additionally, because of the old code-base, the code was not published anywhere, making it impossible for anyone else to make modifications.
+Not really Open Source friendly.
+
+And of course we upgraded other software once in a while .. but it took me a week (as in, 40 hours) to apply patches on all the systems, validate everything was still working, etc etc.
+As I too do this for free, in my free time, the older you get, the more difficult this becomes.
+So it started to slack more and more, up to a point that I found out today that 1 services has been running since 2016.
+This is not okay, in no means what so ever.
+That is a disaster waiting to happen .. it is no longer an "if", but a "when".
```

Comment:
```suggestion
So the work started to stack more and more, up to a point that I found out today that 1 services has been running since 2016.
In no uncertain terms, this is not okay.
It is a disaster waiting to happen; no longer an "if", but a "when".
```